src/Cache/Cache.php

```diff
@@ -199,7 +199,7 @@ protected static function _buildEngine(string $name): void
      * @return \Cake\Cache\CacheEngine
      * @deprecated 3.7.0 Use Cache::pool() instead. This method will be removed in 5.0.
      */
-    public static function engine(string $config)
+    public static function engine(string $config): CacheEngine
```

Comment:
Should this be the psr6 interface hint instead?

Reply:
If I do that I am pretty sure static analyzers will complain about use of methods like `increment()` on the returned instances which are not PSR6 interface methods. So think it would be best to remove the return typehint and update the docblock to union of `CacheInterface` and `CacheEngineInterface`.

Reply:
Done